### PR TITLE
feat: prisma partners listings

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses #3725

- [ ] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR tackles the listings page connection to the new backend on the partners site. It covers all the flows of creating and editing listings. 

It does not cover the following:
- listing and application export is not completed on the backend so that is not included in the PR

There are also a couple of bug fixes found during bug bash:
- Partner user editing the email of a user

## How Can This Be Tested/Reviewed?

After reseeding the prisma backend start up the backend and partner site all listing functionality (minus the above noted) should work.
This includes, but is not limited to:
- /Listings page should load and you can see/paginate the listings
  - Appropriate listings should be shown for each user type
- Create a listing
  - All fields should be filled in and after save should still exist
- Save as draft and publish listing
- Listing status changes (draft, open, closed, etc)
- Edit listing
  - Remove fields, edit fields, add new fields, update assets (images, open houses, etc)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/components/listings/PaperListingForm/sections/BuildingFeatures.tsx
Comment (emilyjablonski) [inline sites/partners/src/components/listings/PaperListingForm/sections/BuildingFeatures.tsx:110]: Is there a ticket for this yet?

File: sites/partners/src/lib/listings/BooleansFormatter.ts
Comment (emilyjablonski) [inline sites/partners/src/lib/listings/BooleansFormatter.ts:61]: Is there a ticket / more information we could add to this comment?

**Review by emilyjablonski — APPROVED** [review]

File: sites/partners/src/lib/listings/BooleansFormatter.ts
Comment (ludtkemorgan) [reply sites/partners/src/lib/listings/BooleansFormatter.ts:61]: ope, this was a comment for myself while working. I still need to figure this part out and will update it here when I do in fact figure it out

File: sites/partners/src/components/listings/PaperListingForm/sections/BuildingFeatures.tsx
Comment (ludtkemorgan) [reply sites/partners/src/components/listings/PaperListingForm/sections/BuildingFeatures.tsx:110]: I can create a ticket for this!

**Review by ColinBuyck — CHANGES_REQUESTED** [review]
I'm seeing the error handling issue from a while ago on this PR. More specifically, when trying to publish an empty listing only the jurisdiction and unit button are highlighted red despite many more fields being required. Below is the state of trying to submit with only the jurisdiction and unit field filled out 

https://github.com/bloom-housing/bloom/assets/53269332/c0c75613-9935-4a46-9f83-6432b52d89f3

**Comment by ludtkemorgan** [comment]
@ColinBuyck good catch! Looks like some additional backend validation was needed for the publish flow which has been added. I also found a bug in our current backend where the error message for the images was not getting shown so I added that as well

**Comment by emilyjablonski** [comment]
@ludtkemorgan I totally could be missing it in the diff, but if we didn't have a test case for the bug Colin called out, could we add one?

**Comment by ColinBuyck** [comment]
Still noticing a couple pieces on the error handling... building address doesn't seem to be highlighting when missing, and the highlighting isn't resetting when filled out 

Screenshot 2023-12-13 at 1 35 26 PM

**Review by ColinBuyck — APPROVED** [review]
LGTM 🍪 once the new merge conflicts are resolved

**Pull request merged**
